packages/timeline-state-resolver/src/integrations/websocketTcpClient/index.ts

```diff
@@ -0,0 +1,154 @@
+import { CommandWithContext, Device, DeviceContextAPI } from '../../service/device'
+import { ActionExecutionResultCode, DeviceStatus, DeviceType, StatusCode, Timeline, TimelineContentTypeWebSocketTcpClient, TSRTimelineContent, WebSocketTCPClientOptions } from 'timeline-state-resolver-types'
+import { WebSocketTcpConnection } from './connection'
+import { WebsocketTcpClientActions } from 'timeline-state-resolver-types'
+
+
+/** this is not an extends but an implementation of the CommandWithContext */
+export interface WebSocketTcpCommand extends CommandWithContext {
+	command: {
+		type: TimelineContentTypeWebSocketTcpClient
+		message: string
+		isBase64Encoded?: boolean
+	}
+	context: string
+  }
+export type WebSocketTcpClientDeviceState = Timeline.TimelineState<TSRTimelineContent>
+
+  export class WebSocketTcpClientDevice extends Device<
+	WebSocketTCPClientOptions,
+	WebSocketTcpClientDeviceState,
+	WebSocketTcpCommand
+> {
+	// Use ! as the connection will be initialized in init:
+	private connection!: WebSocketTcpConnection
+
+	constructor(context: DeviceContextAPI<any>) {
+		super(context)
+	}
+	
+    public async init(options: WebSocketTCPClientOptions): Promise<boolean> {
+		this.connection = new WebSocketTcpConnection(options)
+        await this.connection.connect()
+        return true
+    }
+
+	readonly actions = {
+		[WebsocketTcpClientActions.Reconnect]: async (_id: string) => {
+			await this.connection.connect()
+			return { result: ActionExecutionResultCode.Ok }
+		},
+		[WebsocketTcpClientActions.ResetState]: async (_id: string) => {
+			return { result: ActionExecutionResultCode.Ok }
+		},
+		[WebsocketTcpClientActions.SendWebSocketMessage]: async (_id: string, payload?: Record<string, any>) => {
+			if (!payload?.message) {
+				return { result: ActionExecutionResultCode.Error, response: { key: 'Missing message in payload' } }
+			}
+			await this.sendCommand(payload.message)
```

Comment:
`payload.message` is a string, and sendCommand expects a `WebSocketTcpCommand`
```suggestion
			await this.sendCommand({
				command: {
					type: TimelineContentTypeWebSocketTcpClient.WEBSOCKET_MESSAGE,
					message: payload.message,
					isBase64Encoded: false // this isn't clarified on the payload?
				},
				context: 'Sent through action',
				timelineObjId: ''
			})
```

Reply:
Sending message directly to connecting instead